pkgs/applications/graphics/gscreenshot/default.nix

```
@@ -0,0 +1,86 @@
+{ lib
+, fetchFromGitHub
+, python3Packages
+, gettext
+, gobject-introspection
+, gtk3
+, wrapGAppsHook
+, xdg-utils
+, scrot
+, slop
+, xclip
+, grim
+, slurp
+, wl-clipboard
+, waylandSupport ? true
+, x11Support ? true
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "gscreenshot";
+  version = "3.3.0";
+
+  src = fetchFromGitHub {
+    owner = "thenaterhood";
+    repo = "${pname}";
+    rev = "v${version}";
+    sha256 = "vAJyCamAI8N2InbMYvtzeuPSLkWT3+xA0yyS2Bp5alw=";
+  };
+
+  # needed for wrapGAppsHook to function
+  strictDeps = false;
+  # tests require a display and fail
+  doCheck = false;
+
+  nativeBuildInputs = [ wrapGAppsHook ];
+  propagatedBuildInputs = [ gettext gobject-introspection gtk3 xdg-utils ]
+    ++ lib.optionals waylandSupport [
+      # wayland deps
+      grim
+      slurp
+      wl-clipboard
+    ] ++ lib.optionals x11Support [
+      # X11 deps
+      scrot
+      slop
+      xclip
+    ] ++ (with python3Packages; [
+      pillow
+      pygobject3
+      setuptools
+      (lib.optional (!waylandSupport) xlib) # not needed for wayland build
+    ]);
```

I'm not sure if that works? At least that's not how I usually see it written.

```suggestion
    ] ++ (with python3Packages; [
      pillow
      pygobject3
      setuptools
    ]) ++ lib.optional (!waylandSupport) python3Packages.xlib # not needed for wayland build
    ;
```

---

And about the logic itself, does that mean that `python3Packages` is only needed if _not_ building for wayland? I would imagine it is instead necessary if `x11Support` is turned on.

---

There was a much simpler way to write it, I was being silly. Thanks for pointing it out!